### PR TITLE
Add UTM parameters to Dashboard login buttons

### LINKED ISSUE #7638: Add UTM parameters to Dashboard login buttons [TR-37]
Add UTM parameters to dashboard login buttons so that clicks on these buttons can be tracked and Cypress’s product team can better optimize and drive growth and adoption of the dashboard.

### PR DESCRIPTION
**Pull request by panzarino**
- Closes #7638 

### Additional details

Adds tracking to dashboard login links

### PR Tasks

- [x] Have tests been added/updated?
- [x] Has the original issue been tagged with a release in ZenHub?

### CODE REVIEW

**Review by jennifer-shehane — CHANGES_REQUESTED** [review]
We should probably add a test here, since we're not technically testing that this link in the project setup modal is correctly routing to the dashboard route. 

https://github.com/cypress-io/cypress/blob/develop/packages/desktop-gui/cypress/integration/setup_project_modal_spec.js#L522:L522

**Comment by panzarino** [comment]
@jennifer-shehane Turns out I was putting the UTM parameters on the wrong links, I've updated it now.

I'm fairly unfamiliar as to how to write a full e2e test that will check for full functionality here as these changes span both `desktop-gui` and `server`. I'm also unsure of whether Google Analytics will actually pick up on the UTM parameters when sent through this method, so essentially there needs to be additional testing all around.

File: packages/server/lib/gui/auth.js [hunk 1]
```diff
@@ -58,7 +69,7 @@ const _getOriginFromUrl = (originalUrl) => {
 /**
  * @returns a promise that is resolved with a user when auth is complete or rejected when it fails
  */
-const start = (onMessage) => {
+const start = (onMessage, utm_code) => {
```
Comment (flotwig) [inline packages/server/lib/gui/auth.js:72]: Keep it camelCase until it absolutely doesn't have to be:

```suggestion
const start = (onMessage, utmCode) => {
```

File: packages/server/lib/gui/auth.js [hunk 2]
```diff
@@ -68,6 +79,7 @@ const start = (onMessage) => {
     })
   }
 
+  utm = utm_code
```
Comment (flotwig) [inline packages/server/lib/gui/auth.js:82]:
```suggestion
  utm = utmCode
```

**Review by flotwig — CHANGES_REQUESTED** [review]
> I'm fairly unfamiliar as to how to write a full e2e test that will check for full functionality here as these changes span both `desktop-gui` and `server`. I'm also unsure of whether Google Analytics will actually pick up on the UTM parameters when sent through this method, so essentially there needs to be additional testing all around.

@panzarino I don't think you need a full e2e test for this. Probably just a Cypress test in `desktop-gui` that verifies that the UTM code is sent to the IPC channel, and then a unit test in the `server` `auth_spec` that verifies that the expected URL is generated. You could also add a test in `events_spec` that the IPC event calls `beginAuth` properly.

**Review by flotwig — APPROVED** [review]
nice!

**Review by jennifer-shehane — APPROVED** [review]

**Pull request merged**